app/src/main/java/org/dhis2/utils/customviews/PeriodAdapter.kt

```diff
@@ -0,0 +1,80 @@
+package org.dhis2.utils.customviews
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.RecyclerView
+import java.util.ArrayList
+import java.util.Date
+import java.util.Locale
+import org.dhis2.R
+import org.dhis2.databinding.ItemDateBinding
+import org.dhis2.utils.DateUtils
+import org.hisp.dhis.android.core.period.PeriodType
+
+private class PeriodAdapter(
+    private val periodType: PeriodType,
+    openFuturePeriods: Int
+) : RecyclerView.Adapter<DateViewHolder>() {
+
+    companion object {
+        const val DEFAULT_PERIODS_SIZE = 10
+    }
+
+    private val datePeriods: MutableList<Date>
+    private var onDateSetListener: PeriodDialog.OnDateSet? = null
+    private var lastDate: Date
+
+    init {
+        datePeriods = ArrayList()
+        lastDate = DateUtils.getInstance()
+            .getNextPeriod(periodType, DateUtils.getInstance().today, openFuturePeriods - 1)
+        setDates()
```

Comment:
I see the iniitialization of the old class as :  this.openFuturePeriods = openFuturePeriods;
I have some questions, why are we doing openFuturePeriods - 1 ? Can openFuturePeriods be 0 ?